rollingpin/elasticsearch.py

```diff
@@ -47,20 +47,62 @@ def __init__(self, config, components, hosts, command_line, word):
         self.endpoint = "https://%s/%s/%s" % (base_url, index, index_type)
         self.components = components
 
-    def index_doc(self, doc):
+    def index_doc(self, doc, cb=None):
         """ Index a document in Elasticsearch
         :param doc: dictionary with data to index in ES
+        :param cb: an optional callback that will be added to the
+            ``Agent.request`` deferred.  Will be called with an instance of
+            ``twisted.web.client.Response`` as the first argument.
         :return: Deferred
         """
         with swallow_exceptions("elasticsearch", self.logger):
             agent = Agent(reactor)
             body = JSONBodyProducer(json.dumps(doc))
-            return agent.request(
+            d = agent.request(
                 'POST',
                 self.endpoint,
                 Headers({'User-Agent': ['rollingpin']}),
                 body,
             )
+            if cb:
```

Comment:
if you're just going to return the deferred, why not let the caller do the `addCallback` call? that'd be more flexible (e.g. if the next person wants an `addErrback` call or even two `addCallback` calls) and keeps this api simpler